WeatherApp/Model/WeatherResponse.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  WeatherResponse.swift
+//  WeatherApp
+//
+//  Created on 25/01/2020.
+//  Copyright © 2020 Marcin Wójciak. All rights reserved.
+//
+
+import Foundation
+
+struct WeatherResponse: Decodable {
+    enum CodingKeys: String, CodingKey {
+        case city
+        case dayForecasts = "list"
+    }
+
+    let city: CityInfoResponse
+    let dayForecasts: [DayForecastResponse]
+
+    init(from decoder: Decoder) throws {
+        do {
+            let container = try decoder.container(keyedBy: CodingKeys.self)
+            city = try container.decode(CityInfoResponse.self, forKey: .city)
+            dayForecasts = try container.decode([DayForecastResponse].self, forKey: .dayForecasts)
```

Comment:
To samo co wyżej, ale tutaj dekodujemy tablicę. Dlatego jako typ musimy przekazać typ tablicy czyli `[DayForecastResponse].self`. Trzeba uważać, żeby postawić nawiasy `[]` w dobrych miejsach.

 - `[DayForecastResponse].self` - jest to typ (`.self`) tablicy złożonej z obiektów typu `DayForecastResponse`
 - `[DayForecastResponse.self]` - jest to tablica typów `DayForecastResponse.self` - więcej sensu taka konstrukcja nabiera jeśli `DayForecastResponse` byłoby protokołem, ale my nie o tym :P